### PR TITLE
feat(tier4_planning_launch): declare param path argument

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

set tier4_planning_launch_param_path argument

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: launch/tier4_planning_launch/launch/planning.launch.xml [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
   <arg name="vehicle_info_param_file"/>
 
+  <!-- parameter path -->
+  <arg name="tier4_planning_launch_param_path" default="$(find-pkg-share tier4_planning_launch)/config" description="tier4_planning_launch parmaeter path"/>
```
Comment (kenji-miyake) [inline launch/tier4_planning_launch/launch/planning.launch.xml:7]: typo
```suggestion
  <arg name="tier4_planning_launch_param_path" default="$(find-pkg-share tier4_planning_launch)/config" description="tier4_planning_launch parmeter path"/>
```

File: launch/tier4_planning_launch/launch/planning.launch.xml [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
   <arg name="vehicle_info_param_file"/>
 
+  <!-- parameter path -->
+  <arg name="tier4_planning_launch_param_path" default="$(find-pkg-share tier4_planning_launch)/config" description="tier4_planning_launch parmeter path"/>
```
Comment (kenji-miyake) [inline launch/tier4_planning_launch/launch/planning.launch.xml:7]: sorry, missing `a`. :bow: 

```suggestion
  <arg name="tier4_planning_launch_param_path" default="$(find-pkg-share tier4_planning_launch)/config" description="tier4_planning_launch parameter path"/>
```

**Review by yukkysaito — APPROVED** [review]
LGTM
in near future, is it correct that the .universe and autoware_launcher configs will be synced?

**Comment by takayuki5168** [comment]
> in near future, is it correct that the .universe and autoware_launcher configs will be synced?
Right.
autoware_launch will have config, which is synced with tier4_*_launch in autoware.universe

File: launch/tier4_planning_launch/launch/scenario_planning/scenario_planning.launch.xml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <launch>
   <!-- common param -->
-  <arg name="common_param_path" default="$(find-pkg-share tier4_planning_launch)/config/scenario_planning/common/common.param.yaml"/>
+  <arg name="tier4_planning_launch_param_path"/>
```
Comment (kminoda) [inline launch/tier4_planning_launch/launch/scenario_planning/scenario_planning.launch.xml:3]: May I ask why you did not provide a default parameter here as in planning.launch.xml?
(like `"$(find-pkg-share tier4_planning_launch)/config`)

Comment (takayuki5168) [reply launch/tier4_planning_launch/launch/scenario_planning/scenario_planning.launch.xml:3]: Sorry to talk in Japanese.

- planning.launch.xmlより下のlaunchを呼ぶ人はいないのでは
- この辺もデフォルトパラメータが与えられていなく、このlaunchをコマンドから呼ぶ想定はほぼされてなさそう (このような下のlaunchも呼べるように、ということを合意を取ってできる限りデフォルトパラメータを与える、という方針に決まればデフォルトパラメータ与えていいと思います)
    - https://github.com/autowarefoundation/autoware.universe/blob/5a13fa38f46e87b5423264bafff3851ad229c69b/launch/tier4_planning_launch/launch/scenario_planning/scenario_planning.launch.xml#L8-L9
- デフォルトパラメータを与えすぎるとtypoでミスする可能性あり
    - planning.launch.xmlでは **tier4_planning_launch_param_path**をデフォルトパラメータ付きで、
    - scenario_planning.launch.xmlでは**tier4_planning_launch_path**をデフォルトパラメータ付きで、
    - の場合デフォルトパラメータがあるため実行できてしまう -> typoに気づきにくい

Comment (kminoda) [reply launch/tier4_planning_launch/launch/scenario_planning/scenario_planning.launch.xml:3]: なるほどです。ありがとうございます。
ではlocalizationなども一旦現在のものに準拠して作ります

**Pull request merged**